diff --git a/src/tfa98xx.c b/src/tfa98xx.c
index e4ac25c..a4aaf31 100644
--- a/src/tfa98xx.c
+++ b/src/tfa98xx.c
@@ -145,7 +145,7 @@ static enum tfa_error tfa98xx_write_re25(struct tfa_device *tfa, int value)
 static enum tfa_error tfa98xx_tfa_start(struct tfa98xx *tfa98xx, int next_profile, int vstep)
 {
 	enum tfa_error err;
-	ktime_t start_time, stop_time;
+	ktime_t start_time = 0, stop_time = 0;
 	u64 delta_time;
 
 	if (trace_level & 8) {
