From dae2cc44f76671732b21878de60be766e690a416 Mon Sep 17 00:00:00 2001
From: Richard Hu <richard.hu@technexion.com>
Date: Tue, 1 Jun 2021 09:09:04 +0000
Subject: [PATCH] Use 'pr_warn' instead of 'pr_warning' to adapt to Linux
 5.10.9 kerenl API changes

---
 pn5xx_i2c.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pn5xx_i2c.c b/pn5xx_i2c.c
index b4acbac..7ed8543 100644
--- a/pn5xx_i2c.c
+++ b/pn5xx_i2c.c
@@ -238,7 +238,7 @@ static ssize_t pn54x_dev_read(struct file *filp, char __user *buf,
 			if (gpio_get_value(pn54x_dev->irq_gpio))
 				break;
 
-			pr_warning("%s: spurious interrupt detected\n", __func__);
+			pr_warn("%s: spurious interrupt detected\n", __func__);
 		}
 	}
 
@@ -261,7 +261,7 @@ static ssize_t pn54x_dev_read(struct file *filp, char __user *buf,
 		return -EIO;
 	}
 	if (copy_to_user(buf, tmp, ret)) {
-		pr_warning("%s : failed to copy to user space\n", __func__);
+		pr_warn("%s : failed to copy to user space\n", __func__);
 		return -EFAULT;
 	}
 	return ret;
